Header 컴포넌트 구현

Songwonseok opened the pull request:
## 📕 제목
Header 컴포넌트 구현
관련이슈 #1 #2 #3 

## 📗 작업 내용

- [x] IconButton 컴포넌트 구현
- [x] EditorMenu 컴포넌트 구현
- [x] Logo 컴포넌트 구현

### 실행화면
![image](https://user-images.githubusercontent.com/7006837/100094229-08106480-2e9c-11eb-9451-5e391468a382.png)

## 📘 PR 특이 사항
> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점
- IconSet에 모든 icontype 정리
- IconButton은 color, icontype, size를 속성값으로 줘야함
```
사용예시
<icon-button color="white" icontype="stop" size="20px"> </icon-button>
```
- Logo도 color 속성 사용

Songwonseok committed:
feat: [FE] Logo 컴포넌트 구현

- README에 있는 AUDI 로고 svg로 변환했음
- svg 태그로 렌더링, color 속성으로 색상 변경 가능

- #1

Songwonseok committed:
feat: [FE] IconButton 컴포넌트 구현

- iconSet에 모든 icontype 정리
- color, size, icontype을 속성값으로 줘서 사용
- blade icontype만 viewBox 값을 다르게 설정
- 본문 내용

 #2

Songwonseok committed:
feat: [FE] EditorMenu 컴포넌트 구현

- IconButton 컴포넌트를 사용하여 구현
- file, edit, play, user 순으로 tool 모아둠
- play-tools 같은 경우 아이콘 순서를 audiomass를 참고했음
- 본문 내용

 #3

Songwonseok committed:
feat: [FE] Header 컴포넌트 구현

- Logo, EditorMenu 컴포넌트로 구현
- 본문 내용

Songwonseok committed:
feat: [FE] icontype record_on 추가

- 녹음 진행중 상태 아이콘 추가

Jeongeun-Choi commented on frontend/src/components/IconButton/index.ts:
```diff
@@ -0,0 +1,48 @@
+      this.size = "15px";
+    }
+
+    static get observedAttributes() {
+      // 모니터링 할 속성 이름
+      return ['icontype', 'color', 'size'];
+    }
+    connectedCallback() {
+      this.render();
+    }
+    attributeChangedCallback(attrName, oldVal, newVal) {
+      // 속성이 추가/제거/변경되었다.
+      this[attrName] = newVal;
+    }
```
속성 값을 이런식으로 받아올 수 있네요! 하나 배워갑니당😀

Jeongeun-Choi commented on frontend/src/components/EditorMenu/index.ts:
```diff
@@ -0,0 +1,55 @@
+                </div>
+              </div>
+            `;
+    }
+  };
+  customElements.define('editor-menu', EditorMenu);
+})()
```
파일 끝에는 빈 줄이 들어가야하는게 좋다고 하네요!
참고 : https://velog.io/@doondoony/posix-eol

wooojini replied:
에디터에 EOL 문자가 자동으로 들어가게 설정할 수 있습니다!

Jeongeun-Choi reviewed (approved):
svg 아이콘 찾느라 고생하셨습니당😂 
그리고 customElement에서 속성을 저렇게 쓰는거였군요... 참고 잘 하겠습니다!

wooojini commented on frontend/src/components/EditorMenu/style.scss:
```diff
@@ -0,0 +1,22 @@
+}
+
+.icon-wrap {
+  display:flex;
```
CSS를 정렬해주면 좋을 것 같습니다 ! :-) 
참고 : https://webdesign.tutsplus.com/articles/outside-in-ordering-css-properties-by-importance--cms-21685

wooojini commented on frontend/src/components/IconButton/index.ts:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    static get observedAttributes() {
+      // 모니터링 할 속성 이름
```
주석은 지워주세요! :-)

wooojini reviewed (approved):
고생하셨습니다!! svg 파일이 엄청 많아졌네요!! 리뷰내용은 코멘트 남겼습니다 :-)

pieisland commented on frontend/src/components/Header/index.ts:
```diff
@@ -0,0 +1,26 @@
+
+(() => {
+  const HeaderComponent = class extends HTMLElement {
+
```
빈 엔터가 두 번 이상 안되면 보기 좋을 것 같습니다!

pieisland reviewed (approved):
깔끔하게 잘 만드셨네요! 고생 많으셨어요~

Songwonseok committed:
refactor: [FE] iconSet -> icons로 파일명 변경

- Set이라는 단어가 안맞는것 같아서 iconSet에서 icons로 파일명 변경

Songwonseok committed:
refactor: [FE] scss 정렬

- scss 정렬 도입했음
-참고: https://webdesign.tutsplus.com/articles/outside-in-ordering-css-properties-by-importance--cms-21685

Songwonseok committed:
style: EOL 추가 및 주석 제거

- 소스파일 맨 마지막에 공백 추가
- 주석제거

Songwonseok committed:
refactor: [FE] EditorMen 내부 컴포넌트 모듈화

- 따로 state를 관리해야 될 것 같아서 EditTools, PlaybackTools 컴포넌트
  모듈화
- 모든 버튼에 icontype과 동일한 id값을 속성으로 줬음

 #3

Songwonseok committed:
refactor: [FE] PlaybackTools 컴포넌트 구현

- EditorMenu에서 사용되는 재생관련 도구들을 모듈화 시킴
- 모든 icon-button에는 icontype과 동일한 id값을 줬음
- 본문 내용

Songwonseok committed:
refactor: [FE] EditTools 컴포넌트 구현

- EditorMenu 컴포넌트에서 사용되는 편집관련 도구들 모듈화
- 모든 icon-button에 icontype과 동일한 id를 넣었음

Songwonseok committed:
feat: [FE] components index.js에 EditTools, PlaybackTools import

- 따로 모듈화 시킨 EditTools, PlaybackTools 컴포넌트 적용

Songwonseok committed:
style: [FE] scss 정렬방식 변경

- justify-content도 display:flex와 관련있다고 판단하여 같은 그룹으로
  변경

Songwonseok committed:
style: [FE] icontype camelcase 적용

- 스네이크 케이스 -> 카멜케이스로 변경

Songwonseok committed:
style: [FE] index-> 컴포넌트 명으로 변경

- 개발도중 보기 불편해서 컴포넌트 명으로 변경
- 본문 내용
